### PR TITLE
Feature/deploy-ptri-script

### PR DESCRIPTION
**Pull request by rej156**
yarn hardhat run scripts/deployUsdcMaker.ts --network aurora

it runs ptri, stablemaker then usdcmaker scripts in that order with dependencies (dependencies as in contract deployed addresses as used in the next deploy contract script.)

### CODE REVIEW

**Comment by rej156** [comment]
changed deployUsdcMaker.ts to do the following:

- deploy pTRI
- deploy StableLPMaker w/ pTRI as dep
- set3PoolFeeAddress to StableLPMaker
- deploy UsdcMaker with StableLPMaker as dep

**Comment by rej156** [comment]
constants.ts file needs to be updated post-deployment

File: scripts/constants.ts [hunk 1]
```diff
@@ -59,8 +60,5 @@ export const threePoolSwapUtilsAddress = "0x0564d68404608599e8c567A0bD74F90a942A
 export const threePoolSwapFlashLoanAddress = "0x458459E48dbAC0C8Ca83F8D0b7b29FEfE60c3970";
 export const threePoolLpTokenAddress = "0x87BCC091d0A7F9352728100268Ac8D25729113bB";
 
-
-
-//TODO change this to deployed pTRI contract
-export const pTri = "0x04ED68E80a3c36A61E991a51B656FB733712cec8";
-export const dao = "0xf86119de6ee8d4447C8219eEC20E7561d09816d3";
\ No newline at end of file
+export const pTRI = "0x4F0284Fd38df527CC70d0c53bC0BE67DECCcAF25";
```
Comment (baboobhaiya) [inline scripts/constants.ts:63]: can we delete this?

Comment (rej156) [reply scripts/constants.ts:63]:
```suggestion
```

File: scripts/deployStableTRIStaking.ts [hunk 2]
```diff
@@ -8,7 +8,10 @@ import { ethers, run } from "hardhat";
 const USD_TLP_ADDRESS = "0x87BCC091d0A7F9352728100268Ac8D25729113bB";
 const TRI_ADDRESS = "0xfa94348467f64d5a457f75f8bc40495d33c65abb";
```
Comment (baboobhaiya) [inline scripts/deployStableTRIStaking.ts:9]: import both the addressse from constants

File: scripts/deployStableTRIStaking.ts [hunk 3]
```diff
@@ -22,20 +25,20 @@ async function main(): Promise<void> {
...
   );
-  await factory.deployed();
+  await pTRI.deployed();
 
-  console.log(`pTRI address: ${factory.address}`);
+  console.log(`pTRI address: ${pTRI.address}`);
 
   await run("verify:verify", {
-    address: factory.address,
+    address: pTRI.address,
```
Comment (baboobhaiya) [inline scripts/deployStableTRIStaking.ts:41]: love this :heart:

File: scripts/rewards/deployStableLPMaker.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    usnAddress,
+    usdcAddress,
+    usdtAddress,
+    threePoolLpTokenAddress,
+    dao,
+  ];
+  console.log(...stableLPMakerConstructorArgs);
+
+  const stableLPMaker = await StableLPMaker.connect(deployer).deploy(
+    stableLPMakerConstructorArgs[0],
```
Comment (baboobhaiya) [inline scripts/rewards/deployStableLPMaker.ts:50]: would `...stableLPMakerConstructorArgs` not work?

Comment (rej156) [reply scripts/rewards/deployStableLPMaker.ts:50]: it does, works in the test files due to implied any type but typescript complains here some reason

**Pull request merged**